artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/RefsOperation.java

```diff
@@ -174,7 +174,7 @@ public void afterCommit(final Transaction tx) {
       if (pagedMessagesToPostACK != null) {
          for (MessageReference refmsg : pagedMessagesToPostACK) {
             ((PagedReference)refmsg).removePendingFlag();
-            if (((PagedReference) refmsg).isLargeMessage()) {
+            if (((PagedReference) refmsg).isLargeMessage() && !queue.isNonDestructive()) {
```

Comment:
would you not need to check the ack type? e.g. what if it was expired or killed (i might be having another douche moment )

Reply:
Good point. Fixed it